components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/APIAuthenticationHandler.java

```diff
@@ -615,6 +615,9 @@ protected String getAuthenticatorsChallengeString() {
         StringBuilder challengeString = new StringBuilder();
         if (authenticators != null) {
             for (Authenticator authenticator : authenticators) {
+                if (Boolean.parseBoolean(System.getProperty("removeInternalAPIKeyRealm")) && authenticator instanceof InternalAPIKeyAuthenticator) {
+                    continue;
```

Comment:
Shall we add a null check, if there is a risk of having a parser exception?

Reply:
`Boolean.parseBoolean` handles that null checking. I checked the fix without adding the property. It worked as expected without any issue. Will verify it
<img width="805" alt="Screenshot 2023-08-22 at 11 22 17" src="https://github.com/wso2/carbon-apimgt/assets/43112139/4da3c59d-91f4-48e3-aec3-ecc95f55e02b">